### PR TITLE
feat(ui): add notifications 

### PR DESCRIPTION
**Pull request by Zeeshnhmd**

### CODE REVIEW

File: packages/ui/src/components/shared/NotificationProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	message: ValueOrFunction<Renderable, Toast>;
+	icon?: Renderable;
+	duration?: number;
+	pauseDuration: number;
+	position?: ToastPosition;
+	ariaProps: {
+		role: 'status' | 'alert';
+		'aria-live': 'assertive' | 'off' | 'polite';
+	};
+	style?: React.CSSProperties;
```
Comment (dev-bre) [inline packages/ui/src/components/shared/NotificationProvider.tsx:20]: Let's not allow defining custom styles. I would prefer keep using Tailwind as the only style library.

**Review by dev-bre — CHANGES_REQUESTED** [review]

**Comment by dev-bre** [comment]
Looks great @Zeeshnhmd! I see you have some conflicts and I added a comment for you to look at. Thanks!

File: packages/ui/src/components/shared/NotificationProvider.tsx (hunk 1 above)
Comment (Zeeshnhmd) [reply packages/ui/src/components/shared/NotificationProvider.tsx]: Removed style prop

**Comment by Zeeshnhmd** [comment]
Hey, @dev-bre the conflicts are resolved.

**Review by dev-bre — APPROVED** [review]

**Comment by dev-bre** [comment]
@Zeeshnhmd I pulled the latest main branch and I cannot find how to actually show a notification using your provider. can you please clarify?

**Comment by Zeeshnhmd** [comment]
import { toast } from 'react-hot-toast';

  toast.success('Action successful!', {
        duration: 3000, // Optional, how long the notification should stay (in milliseconds)
      });

**Comment by Zeeshnhmd** [comment]
```
import { toast } from 'react-hot-toast';

  const handleAction = () => {
      toast.success('Action successful!', {
        duration: 3000, // Optional, how long the notification should stay (in milliseconds)
      });
  };

 <button onClick={handleAction}>Perform Action</button>
```

**Comment by Zeeshnhmd** [comment]
Did you get a chance to use this functionality?

**Comment by dev-bre** [comment]
yes thanks. I a going to push an update which is going to use your provider in an hour. 

It's not completely clear to me why you redefined most of the interfaces that are already defined in the d.ts of the library itself. 😅but it is something to consider once I have the notifications in place.

**Pull request merged**